common/perf-test-core/src/main/java/com/azure/perf/test/core/EventPerfTest.java

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.perf.test.core;
+
+import reactor.core.publisher.Mono;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Represents the abstraction of a Performance test class.
+ *
+ * <p>
+ * The performance test class needs to extend this class.
+ * To add any test setup and logic the test class should override {@link EventPerfTest#globalSetupAsync()}
+ * and {@link EventPerfTest#globalCleanupAsync()} methods .
+ * </p>
+ * @param <TOptions> the options configured for the test.
+ */
+public abstract class EventPerfTest<TOptions extends PerfStressOptions> extends PerfTestBase<TOptions> {
```

Comment:
I am fine the name, will just be inconsistent with other languages. But doesn't have to be consistent.
Let's see what @mikeharder  says.

Reply:
In this context, `Event` means that the SDK is "event-based" (raises events when operations are completed) vs "api-based" (method calls return when operations are completed), not related to "Event Hubs".  So `EventPerfTest` is actually more generic than `MessagingPerfTest`, since it could be used for any SDK which uses an event-based pattern.

Reply:
@srnagar any thoughts here ?
or is `EvenPerfTest` fine with you ?